ui-v2/src/components/work-pools/work-pool-queue-page-header/work-pool-queue-page-header.test.tsx

```diff
@@ -0,0 +1,191 @@
+import { QueryClient } from "@tanstack/react-query";
+import {
+	createMemoryHistory,
+	createRootRoute,
+	createRoute,
+	createRouter,
+	RouterProvider,
+} from "@tanstack/react-router";
+import { render, waitFor } from "@testing-library/react";
+import { createWrapper } from "@tests/utils";
+import { describe, expect, it, vi } from "vitest";
+import type { WorkPoolQueue } from "@/api/work-pool-queues";
+import { createFakeWorkPoolQueue } from "@/mocks";
+import {
+	WorkPoolQueuePageHeader,
+	type WorkPoolQueuePageHeaderProps,
+} from "./work-pool-queue-page-header";
+
+// Mock the sub-components that have network behavior
+vi.mock("@/components/work-pools/work-pool-queue-toggle", () => ({
+	WorkPoolQueueToggle: ({ queue }: { queue: WorkPoolQueue }) => (
+		<div data-testid="work-pool-queue-toggle">Toggle for {queue.name}</div>
+	),
+}));
+
+vi.mock("@/components/work-pools/work-pool-queue-menu", () => ({
+	WorkPoolQueueMenu: ({ queue }: { queue: WorkPoolQueue }) => (
+		<div data-testid="work-pool-queue-menu">Menu for {queue.name}</div>
+	),
+}));
```

Comment:
You should prefer using `msw` to mock network calls over mocking out subcomponents.